enhancements/network/smart-nic-ovn-offload.md

```diff
@@ -0,0 +1,398 @@
+
+## Terminology
+
+- SmartNIC: A combined Network Interface Card (NIC) and CPU hardware which
+offloads processing tasks that the system CPU would normally handle
+- SmartNIC node(s): Baremetal servers where SmartNICs are plugged to
+- SmartNIC ARM cores: The embedded aarch64 system running on SmartNIC CPU(s)
+- SR-IOV: Single Root I/O Virtualization, a NIC technology that allows
+isolation of PCIe resources
+- PF: Physical Function, the physical ethernet controller that supports SR-IOV
+- VF: Virtual Function, the virtual PCIe device created from PF
+- VF representor: The port representor of the VF. VF and VF representor are
+created in pairs where VF appears in SmartNIC node and VF representor appears
+in SmartNIC ARM cores
+
+## Motivation
+
+Encryption, Encapsulation, and Virtual switching (IPsec, Geneve, VXLAN, OVS,
+Connection Tracking) to and from containers or VMs are basic building blocks
+of the Hybrid cloud and particularly of OCP. These functions are very compute
+intensive and can consume up to 40 cores from the server’s motherboard CPU
+for terminating 25Gigs of traffic. Terminating 100Gs without hardware offload
+is not feasible at all. Most of our customers are demanding all of the above
+without much use of the CPU on the motherboard of the server.
+
+### Goals
+
+- Only support NVIDIA BlueField2 because other implementations of SmartNIC are
+unknown
+- OVN-Kubernetes control plane including OVS and OVN controller are offloaded
+to SmartNIC, other CNIs are not currently planned
+- SR-IOV pods run on SmartNIC node using VF as pod default route interfaces
+- Manage SmartNIC node offload configurations in OpenShift Operators
+
+### Non-Goals
+
+- Provisioning and managing the OVN-Kubernetes components on SmartNIC ARM cores
+- Support multiple SmartNICs on one OpenShift baremetal worker node
+- Support SmartNICs on OpenShift baremetal master nodes
+
+## Proposal
+
+- User installs SR-IOV Operator (OLM managed)
+- User creates one or more custom MachineConfigPools whose nodes are installed
+with SmartNIC hardware
+- User specifies NodeSelector labels in SR-IOV Network Operator(SNO) CRD to
+match on the custom MachineConfigPools
+- SR-IOV Network Operator renders and attaches SmartNIC MachineConfigs to the
+selected custom MachineConfigPools
+- MachineConfigOperator reboots nodes in custom MachineConfigPool while
+applying MachineConfigs
+- SR-IOV Network Operator configures SmartNIC devices and exposes SR-IOV VFs
+as extended kubernetes resources
+- Nodes in custom MachineConfigPools become ready and start serving SR-IOV pod
+creation request
+- User creates SR-IOV pod by specifying SR-IOV network request in pod
+annotations
+- SR-IOV pod starts with VF device as pod default route interface on SmartNIC
+node
+
+
+### Implementation Details/Notes/Constraints
+
+SmartNIC OVN offload will be enabled with OVN-Kubernetes CNI only.
+It is expected that the following changes will be made:
+
+#### OVN-Kubernetes
```

Comment:
I don't see anywhere where we mention that OvS is running on the ARM as a system service and OVN running as a container. Should we describe that here or below in Design Details? Also add an open question about running Kubelet in stand-alone to manage the OVN Container or is that more in the other enhancement (#738)?

Reply:
Added three bullets in `#### OVN-Kubernetes` to describe how ovn components and ovs are running on SmartNIC, SmartNIC node, non-SmartNIC node.
Added a new section `#### Container lifecycle management on SmartNIC` in `### Open Questions` about RHCOS provision and Kubelet in stand-alone mode on SmartNIC.